skyportal/handlers/api/spectrum.py

```diff
@@ -1149,101 +1158,100 @@ def get(self, obj_id):
                 schema: Error
         """
 
-        obj = Obj.get_if_accessible_by(obj_id, self.current_user)
-        if obj is None:
-            return self.error('Invalid object ID.')
+        with self.Session() as session:
+            obj = session.scalars(
+                Obj.select(session.user_or_token).where(Obj.id == obj_id)
+            ).first()
+            if obj is None:
+                return self.error('Invalid object ID.')
 
-        spectra = (
-            Spectrum.query_records_accessible_by(self.current_user)
-            .filter(Spectrum.obj_id == obj_id)
-            .all()
-        )
+            spectra = session.scalars(
+                Spectrum.select(session.user_or_token).where(Spectrum.obj_id == obj_id)
+            ).all()
 
-        return_values = []
-        for spec in spectra:
-            spec_dict = recursive_to_dict(spec)
-            comments = (
-                CommentOnSpectrum.query_records_accessible_by(
-                    self.current_user,
+            return_values = []
+            for spec in spectra:
+                spec_dict = recursive_to_dict(spec)
+                comments = session.scalars(
+                    CommentOnSpectrum.select(session.user_or_token).where(
+                        CommentOnSpectrum.spectrum_id == spec.id
+                    )
+                ).all()
+                annotations = session.scalars(
+                    AnnotationOnSpectrum.select(session.user_or_token).where(
+                        AnnotationOnSpectrum.spectrum_id == spec.id
+                    )
+                ).all()
+
+                spec_dict["comments"] = sorted(
+                    (
+                        {
+                            **{
+                                k: v
+                                for k, v in c.to_dict().items()
+                                if k != "attachment_bytes"
+                            },
+                            "author": {
+                                **c.author.to_dict(),
+                                "gravatar_url": c.author.gravatar_url,
+                            },
+                        }
+                        for c in comments
+                    ),
+                    key=lambda x: x["created_at"],
+                    reverse=True,
                 )
```

Comment:
As I mentioned in another PR, I'd prefer this as a loop.

Reply:
I think this is another PR if we go that route.